.github/workflows/retag_image.yml

```diff
@@ -47,4 +47,12 @@ jobs:
       - name: Retag image
         run: |
           MANIFEST=$(aws ecr batch-get-image --repository-name ${{ secrets.AWS_RESOURCE_NAME_PREFIX }} --image-ids imageTag=${{ inputs.commit_id }} --query 'images[].imageManifest' --output text)
-          aws ecr put-image --repository-name ${{ secrets.AWS_RESOURCE_NAME_PREFIX}} --image-tag ${{ inputs.image_tag }} --image-manifest "${MANIFEST}"
+          IMAGE_META="$(aws ecr describe-images --repository-name ${{ secrets.AWS_RESOURCE_NAME_PREFIX }} --image-ids imageTag=${{ inputs.commit_id }} --image-ids imageTag=${{ inputs.image_tag }} 2> /dev/null )" #lookup for exisiting image
+          if [[ $? == 0 ]]; then
+              IMAGE_TAGS="$( echo ${IMAGE_META} | jq '.imageDetails[0].imageTags[0]' -r )"
+              echo "image with the version ${{ inputs.commit_id }} and tag ${{ inputs.image_tag }} in the repo ${{ secrets.AWS_RESOURCE_NAME_PREFIX }} already exists. Exiting."
+          else
+              echo "${{ secrets.AWS_RESOURCE_NAME_PREFIX }} ${{ inputs.commit_id }}/$image_tag not found. Tagging $image_tag image."
+              aws ecr put-image --repository-name ${{ secrets.AWS_RESOURCE_NAME_PREFIX}} --image-tag ${{ inputs.image_tag }} --image-manifest "${MANIFEST}"
+              exit 1
+          fi
```

Comment:
I think there may be a logic error.  Don't we want attempt to upload the image and then check if it's already present?

```
if ! aws ecr put-image --repository-name ${{ secrets.AWS_RESOURCE_NAME_PREFIX}} --image-tag ${{ inputs.image_tag }} --image-manifest "${MANIFEST}"; then
    IMAGE_META="$(aws ecr describe-images --repository-name ${{ secrets.AWS_RESOURCE_NAME_PREFIX }} --image-ids imageTag=${{ inputs.commit_id }} --image-ids imageTag=${{ inputs.image_tag }} 2> /dev/null )" #lookup for existing image
   if [[ $? == 0 ]]; then
    echo "Warning: image with the version ${{ inputs.commit_id }} and tag ${{ inputs.image_tag }} in the repo ${{ secrets.AWS_RESOURCE_NAME_PREFIX }} already exists. Exiting."
  else
    # Failed to upload image for some reason other than it was previously tagged with the same tag.
    exit 1
  fi
fi
```